### PR TITLE
Added support for approles

### PR DESCRIPTION
**Pull request by dervolker**
Hello Johan,

First thank you for your good work, that helps us a lot. As we use the approle auth backend with vault, I did add support for it. It would be great if you could merge it.

kind regards, Volker

### CODE REVIEW

File: vault.py [hunk 1]
```diff
@@ -134,16 +151,30 @@ def run(self, terms, inject=None, variables=None, **kwargs):
...
+        return token
+
     def _fetch_token(self, cafile, capath, github_token, url, cahostverify):
+        request_url = urljoin(url, "v1/auth/github/login")
+        req_params = {}
+        req_params['token'] = github_token
+        result = self.__fetch_client_token(cafile, capath, request_url, req_params, cahostverify)
+        return result
+
+    def __fetch_client_token(self, cafile, capath, url, data, cahostverify):
```
Comment (jhaals) [inline vault.py]: One underscore is enough here.

File: vault.py [hunk 2]
```diff
@@ -134,16 +151,30 @@ def run(self, terms, inject=None, variables=None, **kwargs):
...
+        request_url = urljoin(url, approle_role_path)
+        req_params = {
+            'role_id': role_id,
+            'secret_id': secret_id
+        }
+        result = self.__fetch_client_token(cafile, capath, request_url, req_params, cahostverify)
+        token = result['auth']['client_token']
+        return token
+
     def _fetch_token(self, cafile, capath, github_token, url, cahostverify):
```
Comment (jhaals) [inline vault.py]: I like the `_fetch_approle_token` you've added above, please rename this method to `_fetch_github_token`

**Review by jhaals — CHANGES_REQUESTED** [review]
Thanks for the kind words and this pull request!
Two minor changes and this is good to go, nice work 👍

File: vault.py (hunk 2 above)
Comment (dervolker) [reply vault.py]: I've added it right now.

File: vault.py (hunk 1 above)
Comment (dervolker) [reply vault.py]: You're right, just implemented it.

**Pull request merged**